app/controllers/time_clock_periods_controller.rb

```diff
@@ -0,0 +1,64 @@
+class TimeClockPeriodsController < InternalController
+  before_action :set_time_clock_period, only: %i[ show edit update destroy ]
+
+  # GET /time_clock_periods
+  def index
+    @q = policy_scope(TimeClockPeriod).ransack(params[:q])
+    @time_clock_periods = @q.result(distinct: true).page(params[:page])
+  end
+
+  # GET /time_clock_periods/1
+  def show
+    @q = policy_scope(TimeClockPunch).where(time_clock_period: @time_clock_period).ransack(params[:q])
+    @time_clock_punches = @q.result(distinct: true).page(params[:page])
+  end
+
+  # GET /time_clock_periods/new
+  def new
+    @time_clock_period = authorize TimeClockPeriod.new
+    @permissions_keys = TimeClockPeriod.permissions.keys.reject { |key| %w[ added_by_admin added_by_user ].include? key unless current_user.admin }.reject { |key| key == 'added_by_manager' unless current_user.person.manager? }
```

Comment:
This should probably get refactored out into another `before_action` since it's the same thing across all of the actions

Reply:
Done